src/scss/parse.js

```diff
@@ -2503,7 +2503,7 @@ function checkParentSelectorExtension(i) {
   let l;
 
   while (i < tokensLength) {
-    if (l = checkNumber(i) || checkIdent(i)) i += l;
+    if (l = checkNumber(i) || checkIdent(i) || checkTypeSelector(i)) i += l;
```

Comment:
Sorry for such long absence!
Don't you think that `checkInterpolation` will be more suitable here since the part after `&` is not an independent selector but just some part of it?